waspc/src/Wasp/Generator/DbGenerator/Jobs.hs

```diff
@@ -130,56 +141,48 @@ seed projectDir seedName =
 -- Since nothing is passed to stdin, `prisma db execute` just runs an empty
 -- SQL command, which works perfectly for checking if the database is running.
 dbExecuteTest :: Path' Abs (Dir ProjectRootDir) -> J.Job
-dbExecuteTest projectDir =
-  let absSchemaPath = projectDir </> dbSchemaFileInProjectRootDir
-   in runPrismaCommandAsDbJob
-        projectDir
-        (const ["db", "execute", "--stdin", "--schema", SP.fromAbsFile absSchemaPath])
+dbExecuteTest projectRootDir =
+  let absSchemaPath = projectRootDir </> dbSchemaFileInProjectRootDir
+   in runPrismaCommandAsJob
+        projectRootDir
+        ["db", "execute", "--stdin", "--schema", SP.fromAbsFile absSchemaPath]
 
 -- | Runs `prisma studio` - Prisma's db inspector.
 runStudio :: Path' Abs (Dir ProjectRootDir) -> J.Job
-runStudio projectDir = runPrismaCommandAsDbJob projectDir $ \schema ->
-  ["studio", "--schema", SP.fromAbsFile schema]
-
-generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> (String, String) -> JobType -> J.Job
-generatePrismaClient projectDir prismaClientOutputDirEnv jobType =
-  runPrismaCommandAsJobWithExtraEnv jobType envVars projectDir $ \schema ->
-    ["generate", "--schema", SP.fromAbsFile schema]
+runStudio projectRootDir =
+  runPrismaCommandAsJob projectRootDir ["studio", "--schema", SP.fromAbsFile schema]
   where
-    envVars = [prismaClientOutputDirEnv]
+    schema = projectRootDir </> dbSchemaFileInProjectRootDir
 
-runPrismaCommandAsDbJob ::
-  Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
-  J.Job
-runPrismaCommandAsDbJob projectDir makeCmdArgs =
-  runPrismaCommandAsJob J.Db projectDir makeCmdArgs
+generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> J.Job
+generatePrismaClient projectRootDir =
+  runPrismaCommandAsJob projectRootDir args
+  where
+    args =
+      [ "generate",
+        "--schema",
+        SP.fromAbsFile $ projectRootDir </> dbSchemaFileInProjectRootDir
+      ]
 
-runPrismaCommandAsJob ::
-  JobType ->
-  Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
-  J.Job
-runPrismaCommandAsJob jobType projectDir makeCmdArgs =
-  runPrismaCommandAsJobWithExtraEnv jobType [] projectDir makeCmdArgs
+runPrismaCommandAsJob :: Path' Abs (Dir ProjectRootDir) -> [String] -> J.Job
+runPrismaCommandAsJob projectRootDir args =
+  runPrismaCommandAsJobWithExtraEnv [] projectRootDir args
 
 runPrismaCommandAsJobWithExtraEnv ::
-  JobType ->
   [(String, String)] ->
   Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
+  [String] ->
```

Comment:
What was lost with not doing the callback? I believe injection of Schema, right? Meaning that each of those functions now has to provide schema on its own, while before schema was injected?

Purely from the standpoint of having less chance of introducing a mistake due to centralized logic (of schema path), to me what we had before looks superior. It is not a terrible difference, I am sure we can survive with redefining schema again and again, but sounds better not to have to do it. I don't understand what you gained by this change where you removed this callback, you said code is less "jumpy"? What does that mean? Why was code confusing?

Reply:
Why wouldn't such function be concerned with locating schema file? Sounds to me better to have that function figure it out, then every single Prisma Command having to redefine it? But this is very hard thing to discuss, isn't it -> it is up to us to decide how "managed" experience this function provides.

Reply:
By the "jumpy" code, I meant:
I was confused by the callback. Sending a callback as an argument separates the logic into two different places (the callback body and its call site), making the reader have to "jump" between two code locations to realize what's going on.

Sending a value, on the other hand, is simple - the code has a value and uses it.

In short, a higher-order function is a an abstraction that adds complexity and reduces duplication (as most abstractions do). In this case, I don't think the duplication justifies it.

> Purely from the standpoint of having less chance of introducing a mistake due to centralized logic (of schema path), to me what we had before looks superior.

This is true. I agree. But I'd argue we should handle it elsewhere (see below).

> Why wouldn't such function be concerned with locating schema file?

Such a function should, in my opinion, be as dumb as possible. It should take a command, env variables, arguments, and execute the command.

It should'b be constructing command line arguments through a provided callback.

I'd expect to use `runPrismaCommandAsJobWithExtraEnv` for running a command that doesn't require a `schema` argument without having to go through the ceremony of sending a schema to a callback and then ignoring it.

Also, what happens when different commands need different files for arguments (e.g., one needs a schema, the other needs a target directory, etc.). We'd have to update the HOF in the signature to expect all of those. In an extreme case, each passed callback would have to take multiple arguments and ignore most of them.

Furthermore, `runPrismaCommandAsJobWithExtraEnv` might not know how to get all the arguments, which means we'd be constructing one part of the argument array in the caller, and the other part by evaluating the callback (making the code even more "jumpy").

It seems cleaner to simply construct the arguments on the outside and pass them into `runPrismaCommandAsJobWithExtraEnv`. Yes, there's duplication, but only because our functions are simple and all of them only require a schema.

I don't think the callback abstraction occurs naturally. Still, if you have a strong preference for it, I'll change it back.

Reply:
The idea was just that that command manages prisma schema argument, because that was a bit PITA and is almost always needed. Maybe I forgot how this part of code works, but I think you can still pass other arguments, even though the function was taking care of schema? So there is no need to beef the callback up with other stuff if needed, as you described?
In any case, not a biggie, I am ok with this solution also, I don't think it matters much, so all good from my side.

Reply:
Feel free to resolve this comment if you are all good with it.